tripal_chado/src/TripalField/ChadoFieldItemBase.php

```diff
@@ -755,6 +756,194 @@ protected function removeIncompatibleField(string $bundle, string $machine_name)
+   */
+  protected static function discoverLinked(TripalEntityType $bundle, string $field_id, array $field_types,
+      array $field_instances, array $options): array {
+
+    $field_list = [];
+
+    // See if the base table has a foreign key to the field's table
+    // through an intermediate linking table. discover() only
+    // supports linking tables of the standard naming scheme.
+    $possible_linking_tables = [
+      $options['base_table'] . '_' . $options['table'],
+      $options['table'] . '_' . $options['base_table'],
+    ];
+    foreach ($possible_linking_tables as $linking_table) {
+      if ($options['chado']->schema()->foreignKeyExists($linking_table, $options['base_table'])) {
+        $linking_def = $options['chado']->schema()->getForeignKeyDef($linking_table, $options['base_table']);
+        $base_column = array_keys($linking_def['columns'])[0];
+        if ($options['chado']->schema()->foreignKeyExists($linking_table, $options['table'])) {
+          $fk_def = $options['chado']->schema()->getForeignKeyDef($linking_table, $options['table']);
+          $linker_fkey_column = array_keys($fk_def['columns'])[0];
+          // Check for existing fields of this type.
+          if (array_key_exists($options['id'], $field_types)) {
+            // If yes, then add it to the field list.
+            foreach ($field_instances as $instance) {
+              if ($instance->getType() == $options['id']) {
+                $field_list[] = TripalFieldCollection::getFieldArrayFromFieldInstance($instance);
+              }
+            }
+          }
+
+          // If this is new, then create a field entry in the list.
+          if (!$field_list) {
+            $options['linker_table'] = $linking_table;
+            $options['linker_fkey_column'] = $linker_fkey_column;
+            $options['required'] = FALSE;  // Linker fields are never required
+            $options['cardinality'] = -1;  // Linker fields are always unlimited cardinality
+            $field_list[] = self::createFieldEntry($bundle, $options);
+          }
+        }
+      }
+    }
+
+    return $field_list;
+  }
+
+  /**
+   * {@inheritDoc}
+   * @see \Drupal\tripal\TripalField\Interfaces\TripalFieldItemInterface::discover()
+   *
+   * @param array $options
+   *   Specific options from the field's discover() function. Required keys:
+   *     'id' - the field id, e.g. 'chado_organism_type_default'
+   *     'table' - the field's base table, e.g. 'organism'
+   *     'label' - the field's label, e.g. 'Organism'
+   *     'termIdSpace' - The field term's DB
+   *     'termAccession' - The field term's accession
+   *     'description' - A text description for the discovered field
+   */
+  public static function discover(TripalEntityType $bundle, string $field_id, array $field_types,
+      array $field_instances, array $options = []): array {
+
+    // The parent class only initializes an empty array, so we don't need to call it
+    $field_list = [];
```

Comment:
We should call the parent anyway. Right now the parent doesn't add fields but one day it might... debugging would suck if one day the parent does add fields ;-p

Reply:
Yes good advice